app/Router.py

```diff
@@ -13,14 +15,104 @@
 Route = namedtuple('Route', ['host', 'path', 'endpoint'])
 
 
+path_re = re.compile(
+    r"""
+    (\*+)              # wildcard ie: /* or /end* or /start/*/end
+    |/:([a-zA-Z0-9_]+)
+    |/([a-zA-Z0-9_]+)# path variable ie: /user/:id
+    """,
+    re.VERBOSE,
+)
+
+# this is used to match "simple paths, and attempt to
+# match without regex to speed up processing
+path_matcher = re.compile('^([A-Za-z0-9-._~()\'!@,;_/]+)$')
+
+wild_matcher = re.compile(
+    "^(?P<wildcard>[A-Za-z0-9-._~()\\'!*:@,;_/]+)?(?P<endpath>/)?$"
+)
+
+match_wild_re = r'(?P<wildcard>[A-Za-z0-9-._~()\'!*:@,;_/]+)?'
+
+
+def match_var_re(var_name):
+    return r'/(?P<%s>[A-Za-z0-9-._~()\'!*:@,;]+)?' % var_name
+
+
+def build_route_matcher(path):
+    """
+    Parses the provided path and returns the regular expression
+    described by the path.
+    """
+
+    if path is None:
+        return None
+
+    match_regex_parts = []
+    used_names = set()
+
+    if not path or (path[0] != '/' and path[0] != '*'):
+        raise ValueError('path must begin with / or *')
+
+    # If we are attempting to match a simple path, let's go
+    # ahead and return the path. This will tell the matcher
+    # to avoid regex matching if the path is the same.
+
+    path_exact = path_matcher.match(path)
+    if path_exact is not None and \
+            path_exact[0] is path:
+        return path
+
+    if path == '/*' or path == '*':
+        match_regex_parts.append(match_wild_re)
+    else:
+        count = 0
+        for m in path_re.findall(path):
+            count = count + 1
+            if m[0] is not '':
+                # wildcard
+
+                if match_wild_re in match_regex_parts:
+                    raise ValueError('wildcard * used more than once')
+
+                match_regex_parts.append(match_wild_re)
+            elif m[1] is not '':
+                # path variables
+
+                var = m[1]
+
+                if var.lower() == 'wildcard':
+                    raise ValueError(
+                        'path variable name :wildcard is reserved'
+                    )
+
+                if var in used_names:
+                    raise ValueError(
+                        'path variable %r used more than once.' % var
+                    )
+                match_regex_parts.append(match_var_re(var))
+                used_names.add(var)
+            elif m[2] is not '':
+                # regular path
+                match_regex_parts.append(f'/{m[2].replace("/", "")}')
+
+    # we utilize this so we do not cause any errors
+    match_regex_parts.append(r'(?P<endpath>/)?')
+
+    return re.compile(r'^%s$' % r''.join(match_regex_parts))
+
+
 def dict_decode_values(_dict):
```

Comment:
According to PEP8 it would be:

```suggestion
def dict_decode_values(dict_):
```